### PR TITLE
Reading exchange name from endpoint config implementation

### PR DESCRIPTION
**Pull request by kaustuk**
**Proposed changes**:
implemented following feature request
- fix https://github.com/RasaHQ/rasa/issues/7848
- fix https://github.com/RasaHQ/rasa/issues/7889
Here for pika event broker added 1 extra parameter called `exchange_name` which help user to configure exchange name from `endpoints.yml` file.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by sara-tagger** [comment]
Thanks for submitting a pull request 🚀 @m-vdb will take a look at it as soon as possible ✨

File: rasa/core/brokers/pika.py [hunk 1]
```diff
@@ -62,6 +63,8 @@ def __init__(
             connection_attempts: Number of attempts for connecting to RabbitMQ before
                 an exception is thrown.
             retry_delay_in_seconds: Time in seconds between connection attempts.
+            exchange_name: Exchange name to which the queues binds to.
+            If nothing is mention then Default exchange name would be used
```
Comment (wochinge) [inline rasa/core/brokers/pika.py:67]:
```suggestion
                If nothing is mention then the default exchange name would be used
```

File: rasa/core/brokers/pika.py [hunk 2]
```diff
@@ -146,7 +150,9 @@ async def connect(self) -> None:
         logger.info(f"RabbitMQ connection to '{self.host}' was established.")
 
         channel = await self._connection.channel()
-        logger.debug("RabbitMQ channel was opened. Declaring fanout exchange.")
+        logger.debug(
+            f"RabbitMQ channel was opened. Declaring fanout exchange. exchange name: {self.exchange_name}"
```
Comment (wochinge) [inline rasa/core/brokers/pika.py:154]: what do you think?
```suggestion
            f"RabbitMQ channel was opened. Declaring fanout exchange '{self.exchange_name}'."
```

File: data/test_endpoints/event_brokers/pika_endpoint.yml [hunk 3]
```diff
@@ -8,3 +8,4 @@ event_broker:
 #   you may supply more than one queue to publish to
 #   - queue-2
 #   - queue-3
+  exchange_name: exchange
```
Comment (wochinge) [inline data/test_endpoints/event_brokers/pika_endpoint.yml:11]:
```suggestion
  exchange_name: exchange

```

**Review by wochinge — COMMENTED** [review]
Thanks for the contribution! Would you mind adding a [changelog](https://github.com/RasaHQ/rasa/blob/d37631e871bdced59f1c580ac710f7b1ba35cca0/changelog/README.md) for this?

**Comment by kaustuk** [comment]
> Thanks for the contribution! Would you mind adding a [changelog](https://github.com/RasaHQ/rasa/blob/d37631e871bdced59f1c580ac710f7b1ba35cca0/changelog/README.md) for this?

Hi @wochinge, thanks for reviewing PR so fast. Want to ask what should be the issue type for changelog? I think it should be `improvement` But can you still confirm it once. 

Sorry for asking silly question, but this is my first time.

**Comment by wochinge** [comment]
`improvement` sounds good!

File: rasa/core/brokers/pika.py (hunk 2 above)
Comment (kaustuk) [reply rasa/core/brokers/pika.py:154]: Yeah make sense. Will make changes as suggested

File: rasa/core/brokers/pika.py (hunk 1 above)
Comment (kaustuk) [reply rasa/core/brokers/pika.py]: Done

File: changelog/7848.improvement.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Added support to configure `exchange name` for [pika event broker](event-brokers.mdx#pika-event-broker)
```
Comment (wochinge) [inline changelog/7848.improvement.md:1]:
```suggestion
Added support to configure `exchange_name` for [pika event broker](event-brokers.mdx#pika-event-broker).
```

File: rasa/core/brokers/pika.py [hunk 5]
```diff
@@ -62,6 +63,8 @@ def __init__(
             connection_attempts: Number of attempts for connecting to RabbitMQ before
                 an exception is thrown.
             retry_delay_in_seconds: Time in seconds between connection attempts.
+            exchange_name: Exchange name to which the queues binds to.
+                If nothing is mention then Default exchange name would be used
```
Comment (wochinge) [inline rasa/core/brokers/pika.py:67]:
```suggestion
                If nothing is mentioned then the default exchange name would be used.
```

**Review by wochinge — APPROVED** [review]
Thanks for this contribution 💯

**Pull request merged**